clouddriver-kubernetes/src/main/groovy/com/netflix/spinnaker/clouddriver/kubernetes/v1/api/KubernetesApiConverter.groovy

```diff
@@ -809,8 +809,20 @@ class KubernetesApiConverter {
   static ReplicaSet toReplicaSet(ReplicaSetBuilder serverGroupBuilder,
                                  DeployKubernetesAtomicOperationDescription description,
                                  String replicaSetName) {
-
-    def targetSize = description.targetSize ?: description.capacity?.desired
+    /*
+    The elvis operator before was not evaluating truthiness correctly for a value of 0.
+    A User could and may want to deploy an update ReplicaSet of 0 and Scale it later.  Or perhaps it's
+    bundled in a Spinnaker Pipeline with multiple Server Groups and one of the Server Groups is scaled to 0.
+    If the User is implementing 'useSourceCapacity' we would want that 0 value respected and not overwritten with a
+    value of 1, which will happen if the targetSize is represented by a null value at this juncture in the code.
+
+    Validation around targetSize already happens at the Create Server Group level.  This is just properly casting the
+    description.targetSize.
+     */
+    Integer targetSize = description.targetSize.toInteger()
+    if (targetSize < 0) {
```

Comment:
What happens if `targetSize` is null? Does `description.targetSize.toInteger()` return `-1` in that case? Just want to make sure we preserve the original behavior as well.

Reply:
targetSize cannot be null - that's part of the Server Group Validation.